src/vs/workbench/parts/debug/electron-browser/debugHover.ts

```diff
@@ -104,6 +104,49 @@ export class DebugHoverWidget implements editorbrowser.IContentWidget {
 		return this.domNode;
 	}
 
+	private getHoveredExpression(lineContent: string, range: Range) : Range {
+		let matchingExpression = undefined;
+		let startOffset = 0;
+
+		// Some example supported expressions: myVar.prop, a.b.c.d, myVar?.prop, myVar->prop, MyClass::StaticProp, *myVar
+		// Match any character except a set of characters which often break interesting sub-expressions
+		let expression: RegExp = /([^()\[\]{}<>\s+\-/%~#^;=|,`!]|\->)+/g;
```

Comment:
doesn't this filter out template/generic types? For instance, in C++, an expression like this should be valid:
CSeafood<Squid>::Swim  where CSeafood is a template instance with template parameter Squid.